arrow/src/main/java/org/apache/iceberg/arrow/vectorized/VectorHolder.java

```diff
@@ -140,14 +141,16 @@ public static class ConstantVectorHolder<T> extends VectorHolder {
     private final int numRows;
 
     public ConstantVectorHolder(int numRows) {
+      super(new NullVector("_dummy_", numRows), null, null);
       this.numRows = numRows;
       this.constantValue = null;
     }
 
     public ConstantVectorHolder(Types.NestedField icebergField, int numRows, T constantValue) {
-      super(icebergField);
+      super(new NullVector(icebergField.name(), numRows), icebergField, new NullabilityHolder(numRows));
       this.numRows = numRows;
       this.constantValue = constantValue;
+      this.vector = new NullVector(icebergField.name(), numRows);
```

Comment:
I don't think we should be creating two instances of `NullVector`. Either use the one you pass via `super(...)` or this one, but not both

Reply:
Not sure how I missed that because this code doesn't even compile. I've removed this line entirely.